### PR TITLE
Fix #403

### LINKED ISSUE #403: References occasionally get sent with invalid attributes
Not sure why this happens, will look into it.

```

fhq-0 (err):     at /var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:162:20
fhq-0 (err):     at /var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:230:13
fhq-0 (err):     at _arrayEach (/var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:81:9)
fhq-0 (err):     at _each (/var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:72:13)
fhq-0 (err):     at Object.async.forEachOf.async.eachOf (/var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:229:9)
fhq-0 (err):     at Object.async.forEach.async.each (/var/www/flairhq/node_modules/waterline/node_modules/async/lib/async.js:206:22)
fhq-0 (err):     at [object Object].Validator.validate (/var/www/flairhq/node_modules/waterline/lib/waterline/core/validations.js:207:9)
fhq-0 (err):     at /var/www/flairhq/node_modules/waterline/lib/waterline/query/validate.js:42:25
fhq-0 (err): 
fhq-0 (err): Invalid attributes sent to Reference:
fhq-0 (err):  • number
fhq-0 (err):    • `undefined` should be a integer (instead of "NaN", which is a number)
fhq-0 (err):    • "required" validation rule failed for input: NaN

```

### PR DESCRIPTION
**Pull request by not-an-aardvark**
~~Error 404: Pull request not found.~~

tl;dr We check to make sure the number field is valid when references are added, but not when they're edited.

I also added clientside checks for it.

### CODE REVIEW

File: assets/js/sharedClientFunctions.js [hunk 1]
```diff
@@ -42,11 +42,15 @@ module.exports = {
         ref.user2 = ref.user2.slice(3);
       }
       if (ref.user2 === ($scope.user.name)) {
-        $scope.addRefError = "Don't put your own username there.";
+        $scope.editRefError = "Don't put your own username there.";
         return;
       }
       if (($scope.type !== "giveaway" && $scope.type !== "misc") && !regexpUser.test(ref.user2)) {
-        $scope.addRefError = "Please put a username on its own, or in format: /u/username. Not the full url, or anything else.";
+        $scope.editRefError = "Please put a username on its own, or in format: /u/username. Not the full url, or anything else.";
```
Comment (YaManicKill) [inline assets/js/sharedClientFunctions.js]: We should really try and combine most of the logic in this and add ref. This isn't the first time something has been fixed in 1 and not the other.
